lib/yoti/dynamic_share_service/policy/dynamic_policy.rb

```diff
@@ -93,81 +93,87 @@ def with_wanted_attribute(attribute)
 
       #
       # @param [String] name
-      #
-      def with_wanted_attribute_by_name(name)
-        attribute = WantedAttribute.builder.with_name(name).build
+      # @param [Hash] constraints
+      #
+      def with_wanted_attribute_by_name(name, constraints: nil)
+        attribute_builder = WantedAttribute.builder.with_name(name)
+        constraints&.each do |constraint|
+          attribute_builder.with_constraint constraint
+        end
+        attribute = attribute_builder.build
         with_wanted_attribute attribute
       end
 
-      def with_family_name
-        with_wanted_attribute_by_name Attribute::FAMILY_NAME
+      def with_family_name(options = {})
+        with_wanted_attribute_by_name Attribute::FAMILY_NAME, options
```

Comment:
`options` is just constraints currently isn't it? It might be confusing if we later add in other options. I think it would be better to call this `constraints` throughout

Reply:
The intention is that if we add other options they'd be passed in this one map. Naming it constraints would increase the number of places this would need to be updated.

Reply:
Ok, happy to leave as is to use the Ruby automagic then